sklearn/model_selection/_split.py

```diff
@@ -632,6 +644,12 @@ def split(self, X, y, groups=None):
 
         test : ndarray
             The testing set indices for that split.
+
+        Note
+        ----
+        Multiple calls to the ``split`` method will not return identical
+        training or testing sets unless ``random_state`` is set to an integer
```

Comment:
I am really unable to recollect why I did that :@ :@ :@ Arghh. I guess it was done so different split calls would produce same split? But I guess that was vetoed down (see: https://github.com/scikit-learn/scikit-learn/pull/7935).

I'll revert for now. I can revert when someone complains.